### PR TITLE
Added Internal Test & Updated other templates

### PR DESCRIPTION
**Pull request by andrewballantyne**
Closes: #1325 

### Test Template

Screenshot 2023-07-04 at 12 39 36 PM

### Bug Template

Screenshot 2023-07-04 at 12 38 26 PM
Screenshot 2023-07-04 at 12 38 35 PM
Screenshot 2023-07-04 at 12 38 46 PM

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -14,6 +14,27 @@ body:
...
+    id: deploy-type
+    attributes:
+      label: Deploy type
+      description: |
+        How did you deploy the Dashboard?
+
+        Eg. what is the repo location, `odh-manifests` or `odh-dashboard`?
+      multiple: false
+      options:
+        - Using an OpenDataHub main version (eg. `v1.6.0`)
```
Comment (lucferbux) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:27]: mmmm this is great, it will allow us to discriminate some issues.

**Review by lucferbux — APPROVED** [review]
/lgtm

The testing section is more or less what I had in mind, I think it covers all the needed fields to log future testing efforts.

**Comment by andrewballantyne** [comment]
> The testing section is more or less what I had in mind, I think it covers all the needed fields to log future testing efforts.

@Gage We'll need to update https://github.com/opendatahub-io/odh-dashboard/blob/main/docs/architecture.md#testing-structure from your internal google doc -- all things should be exposed so we can have proper open-source collaboration.

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 1 above)
Comment (andrewballantyne) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:27]: Yeah, might need to improve this over time -- but at least it has some information.

Probably need some details how to find this information maybe 🤔

**Review by Gkrumbach07 — APPROVED** [review]

**Pull request merged**